### PR TITLE
Breaking: Remove deprecated features.

### PR DESCRIPTION
**Pull request by notlee**
Removes features recently deprecated:
https://github.com/Financial-Times/o-brand/pull/20
https://github.com/Financial-Times/o-brand/pull/20

### CODE REVIEW

File: scss/mixins.scss [hunk 1]
```diff
@@ -20,262 +17,106 @@ $_o-brand-default-not-set-warning: false !default; // Has the default not set wa
     @if type-of($brand) != 'string' or $brand == '' {
         @error 'Could not set brand "#{$brand}" for component "#{$component}". A brand of type string must be given.';
     }
-    // Define brand configuration.
-    @if not _oBrandIsDefined($component, $brand) {
-        $result: _oBrandUpdateConfig($component, $brand, $config);
+    // Do not allow a component brand to be overridden.
+    @if _oBrandIsDefined($component, $brand) {
+        @error 'Brand #{$brand} is already defined for #{$component}.';
```
Comment (JakeChampion) [inline scss/mixins.scss]: 👍

File: scss/mixins.scss [hunk 2]
```diff
@@ -20,262 +17,106 @@ $_o-brand-default-not-set-warning: false !default; // Has the default not set wa
     @if type-of($brand) != 'string' or $brand == '' {
         @error 'Could not set brand "#{$brand}" for component "#{$component}". A brand of type string must be given.';
     }
-    // Define brand configuration.
-    @if not _oBrandIsDefined($component, $brand) {
-        $result: _oBrandUpdateConfig($component, $brand, $config);
+    // Do not allow a component brand to be overridden.
```
Comment (JakeChampion) [inline scss/mixins.scss]: I think having a list of brands to compare against will help with validating `$brand`, we will catch situations such as a leading or trailing space (`$brand: 'masterbrand ';`)

File: scss/mixins.scss [hunk 3]
```diff
@@ -20,262 +17,106 @@ $_o-brand-default-not-set-warning: false !default; // Has the default not set wa
     @if type-of($brand) != 'string' or $brand == '' {
```
Comment (JakeChampion) [inline scss/mixins.scss]: I think we should split these error messages into more targeted error messages:
https://github.com/Financial-Times/o-brand/blob/f8911dc95e21ea11a7650e5747cb4257c3c64549/scss/mixins.scss#L14-L19

A specific error message for when the component is an empty string and another for when the brand is an empty string.

File: scss/mixins.scss [hunk 4]
```diff
@@ -290,122 +131,33 @@ $_o-brand-default-not-set-warning: false !default; // Has the default not set wa
     }
     // Validate brand variables and supported variants.
     $variables: map-get($config, 'variables');
-    $supports-variants: if(map-get($config, 'settings'), map-get($config, 'settings'), map-get($config, 'supports-variants'));
+    $supports-variants: map-get($config, 'supports-variants');
     @if $variables and (type-of($variables) != 'map' and type-of($variables) != 'list' ) {
         @error '#{$errorMessage}. Config key "variables" should be a map but is of type #{type-of($variables)}.';
```
Comment (JakeChampion) [inline scss/mixins.scss]: The check here makes me think a `list` is fine to use but the error message specifically states that a `map` should be used, which is it?

Comment (JakeChampion) [reply scss/mixins.scss]: When the configuration is invalid, we should include a link to the brand configuration documentation in the error message.

File: scss/mixins.scss (hunk 1 above)
Comment (JakeChampion) [reply scss/mixins.scss]: I'm not sure when reconfiguring a brand for a component would happen, brand configuration for a component has always been done within the component's sass itself.

File: scss/mixins.scss [hunk 5]
```diff
@@ -20,262 +17,106 @@ $_o-brand-default-not-set-warning: false !default; // Has the default not set wa
...
-            $brand-variables: map-get($original-config, 'variables');
-            @if not map-has-key($brand-variables, $from) {
-                @return null;
-            }
+/// @param {map|string} $from Get variable values from a defined brand variant or from a map of variables.
+/// @return {string | number | color | bool | null}
+@function oBrandGet($component, $variables, $from: null) {
+    // Validate variant.
+    @if type-of($from) != 'string' and type-of($from) != 'map' and type-of($from) != 'null' {
+        @error 'Cound not check support for variant "#{$from}", expecting a string, map, or null but found "#{type-of($from)}".';
```
Comment (JakeChampion) [inline scss/mixins.scss]: `Cound` should be `Could`

File: scss/mixins.scss [hunk 6]
```diff
@@ -20,262 +17,106 @@ $_o-brand-default-not-set-warning: false !default; // Has the default not set wa
...
-        @if type-of($value) == 'map' {
-            @if type-of($key) == 'list' and $o-brand-debug-mode {
-                @warn 'Defining variants as lists is deprecated. Use strings instead. Variant "#{$key}" of brand "#{$brand}" for component "#{$component}".';
-            }
-            $key: _oBrandNormaliseVariant($key);
-        }
-        $sorted-variables: map-merge($sorted-variables, ($key: $value));
+    // Check support for a pre-defined variant.
+    @if type-of($variant) != 'string' {
+        @error 'Cound not check support for variant "#{$variant}", expecting a string, map, or null but found "#{type-of($variant)}".';
```
Comment (JakeChampion) [inline scss/mixins.scss]: `Cound` should be `Could`

**Comment by JakeChampion** [comment]
This is really good work and much less intimidating for me to read and simpler to comprehend than the previous solutions. Excellent work on this @notlee 👏

File: scss/mixins.scss (hunk 4 above)
Comment (notlee) [reply scss/mixins.scss]: An empty map is a list `()`. Added a comment to clarify.

**Pull request merged**